Sources/Automerge/Document.swift

```diff
@@ -946,6 +946,58 @@ public final class Document: @unchecked Sendable {
         }
     }
 
+    /// Generates patches between two points in the document history.
+    ///
+    /// Use:
+    /// ```
+    /// let doc = Document()
+    /// doc.difference(to: doc.heads())
+    /// ```
+    /// - Parameters:
+    ///   - from: heads at beginning point in the documents history.
+    ///   - to: heads at ending point in the documents history.
+    /// - Note: `from` and `to` do not have to be chronological. Document state can move backward.
+    /// - Returns: The difference needed to produce a document at `to` when it is set at `from` in history.
+    public func difference(from before: Set<ChangeHash>, to after: Set<ChangeHash>) -> [Patch] {
+        sync {
+            let patches = self.doc.wrapErrors { doc in
+                doc.difference(before: before.map(\.bytes), after: after.map(\.bytes))
+            }
+            return patches.map { Patch($0) }
+        }
+    }
+
+    /// Generates patches **from** a given point in the document history.
+    ///
+    /// Use:
+    /// ```
+    /// let doc = Document()
+    /// doc.difference(from: doc.heads())
+    /// ```
+    /// - Parameters:
+    ///     - from: heads at beginning point in the documents history.
+    /// - Note: `from` do not have to be chronological. Document state can move backward.
+    /// - Returns: The difference needed to produce current document given an arbitrary
+    /// point in the history.
+    public func difference(from lhs: Set<ChangeHash>) -> [Patch] {
```

Comment:
```suggestion
    public func difference(since lhs: Set<ChangeHash>) -> [Patch] {
```

Using the parameter name `from` or `to` implies directional ordering, and since that is irrelevant (I'm basing that on your comment here) and one side of this function is always the current heads(), that implies this is a convenience function to look back to a prior state. Along those lines `since` distinguishes it as such, implying more of what its doing. Along those lines, I'd drop the replica of this function that reverses the call order if that really is irrelevant.